### PR TITLE
Add Mining Research Points and techgates mining research behind it

### PR DESCRIPTION
**Pull request by redmoogle**
# General Documentation

### Intent of your Pull Request

Kinda like #10790 but done differently

Instead of messing with cutters and all of that. It introduces a new point type `Mining Research` which is 1 Research Point per 2 Mining Points

To get the basic plasmacutter you spend 1250 points. which is 2500 MP which translate to ~2-3 loads of mining satchels
And also moves all of the pKa mods to their own semi-cheap node

### Why is this change good for the game?

Encourages a more friendly science/mining interaction as science has to research it but mining has to well. mine.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
1 New technode, New research point gotten by mining

### What general grouping does this PR fall under? 
Mining/Research Rebalance

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Normal mining node cost 1250 MRP:tm: points
Adv Mining Node costs 4000 MRP points
Proto-Kinetic Mods cost 1500 MRP points

# Changelog

:cl:  
rscadd: Add Mining Research Points
tweak: Techgate mining research behind it
/:cl:

### CODE REVIEW

**Comment by Jarod1200** [comment]
MRP points. Moogle has finally coded in MRP

**Comment by ToGWtF** [comment]
So will this time-gate mining tools behind them mining first?

I dont see this as a "Friendly interaction" but rather slowing down mining.

**Comment by redmoogle** [comment]
> 
> 
> So will this time-gate mining tools behind them mining first?
> 
> I dont see this as a "Friendly interaction" but rather slowing down mining.

They're roughly the same time really, ~5-10 minutes of work at most, except for the miners dont sit on their asses. But at the same time it still keeps research as the way of researching the tech but now they have to get their own points

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
BE READY FOR BUGS!

**Comment by bowlofcereal** [comment]
>make miners mine
>lock mining tools behind their own research that requires miners to mine to get too
>miners now only have a crusher or unupgraded KA to mine with
>to get better things, miners are now more likely to go after lavaland loot, the whole thing people bitch about them doing in the  first place

The only thing this has done is make mining techs longer to get and miners less likely to mine because they can't get cutters. Could you at the very least put a plasmacutter in the mining vendor for a specific point quota to actually make an incentive to click on rocks?
